pkg/users/group-create-dialog.js

```diff
@@ -0,0 +1,170 @@
+                       validated={(errors && errors.id) ? "error" : "default"}
+                       isStack
+                       fieldId="groups-create-id">
+                <Flex spaceItems={{ default: 'spaceItemsSm' }} alignItems={{ default: 'alignItemsCenter' }}>
+                    <Checkbox id="new-group-use-custom-id"
+                              label={_("Use specific ID")}
+                              isChecked={user_specified_id_enabled} onChange={checked => change("user_specified_id_enabled", checked)} />
+                    <FlexItem spacer={{ default: 'spacerLg' }}>
+                        <Popover bodyContent={_("When not specifying the ID the default is to use the smallest ID value greater than or equal to GID_MIN and greater than every other group.")}
+                                 showClose={false}>
+                            <HelpIcon />
+                        </Popover>
+                    </FlexItem>
+                </Flex>
+                {user_specified_id_enabled && <TextInput id="groups-create-id"
+                           validated={(errors && errors.id) ? "error" : "default"}
+                           value={id} onChange={value => change("id", value)} />}
+            </FormGroup>
+        </Form>
+    );
+}
+
+function is_valid_char_name(c) {
+    return (c >= 'a' && c <= 'z') ||
+        (c >= 'A' && c <= 'Z') ||
+        (c >= '0' && c <= '9') ||
+        c == '.' || c == '_' || c == '-';
+}
+
+function validate_name(name, groups) {
+    if (!name)
+        return _("No user name specified");
+
+    for (let i = 0; i < name.length; i++) {
+        if (!is_valid_char_name(name[i]))
+            return _("The group name can only consist of letters from a-z, digits, dots, dashes and underscores.");
+    }
+
+    for (let k = 0; k < groups.length; k++) {
+        if (groups[k].name == name)
+            return _("A group with this name already exists");
+    }
+
+    return null;
+}
+
+function validate_id(id) {
+    if (!id || isNaN(id))
+        return _("No valid id specified");
+}
+
+export function group_create_dialog(groups, setGroupsCardExpanded) {
+    let dlg = null;
+    const state = {
+        name: "",
+        id: "",
+    };
+    let errors = { };
+
+    function change(field, value) {
+        state[field] = value;
+        errors = { };
+
+        update();
+    }
+
+    function validate(name, id, user_specified_id_enabled) {
+        const errs = { };
+
+        errs.name = validate_name(name, groups);
+        if (user_specified_id_enabled)
+            errs.id = validate_id(id);
+        errors = errs;
+
+        return !has_errors(errs);
+    }
+
+    function create(name, id, user_specified_id_enabled) {
+        const valid = validate(name, id, user_specified_id_enabled);
+        if (valid) {
+            let group_add_cmd = ["/usr/sbin/groupadd", name];
+
+            if (user_specified_id_enabled)
+                group_add_cmd = [...group_add_cmd, "-g", id];
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18136-20230109-075233-91ffdc17-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)